File: src (cpp)/animal/pisces/barracuda.cpp
#include "barracuda.h"

Barracuda::Barracuda(int _x, int _y, int _weight) : def_ID(15),
                                                    def_ratio_meat(60),
                                                    def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = false;
  is_water_animal = true;
  is_air_animal = false;
  weight = _weight;
}
void Barracuda::Interact() {
  std::cout << "The barracuda completely ignores you" << std::endl;
}

File: src/virtualzoo/animal/pisces/Barracuda.java
package virtualzoo.animal.pisces;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Barracuda mendefinisikan atribut untuk barracuda.
 *
 * @author Holy Lovenia - 13515113
 * @version 3.0
 * @since 3.0
 */
public class Barracuda extends Pisces {

  /** <p>
   * Constructor
   *
   * Menciptakan barracuda dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat barracuda.
   */
  public Barracuda(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 60);
    id = 15;
    position.setLocation(locX, locY);
    isLandAnimal = false;
    isWaterAnimal = true;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan barracuda.
   */
  public void interact() {
    System.out.println("The barracuda completely ignores you");
  }
}
